vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp

```diff
@@ -242,9 +246,40 @@ ReactInstanceWin::ReactInstanceWin(
   // OnInstanceCreated event is raised only after the internal react-native instance is ready and
   // it starts handling JS queue work items.
   m_whenCreated.SetValue();
+
+  if (options.EnableDefaultCrashHandler()) {
+    CrashManager::SetupCustomHander();
+  }
+
+  {
```

Comment:
Isn't this the bit that should be controlled by the instance settings?

Reply:
I think there is value in tracking all active instances somewhere, but I added a check for the opt-in setting in InstanceCrashHandler.